[Store] : 1차 리팩토링

ybon1107 opened the pull request:
## PR Type
어떤 변경 사항이 있나요?

- [ ] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Summary : 변경 내용
- **as-is**
  - 가게 open/close 두가지 상태를 고려한 설계

- **to-be**
  - 가게 상태 추가를 고려한 command 패턴으로 리팩토링

## Issue Number
- close #73 

## Etc


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added a consolidated endpoint for changing store status (open/close) via a single API with a status parameter.
  - Introduced new error handling for invalid status values and enum conversion errors.
  - Added a new endpoint for reserving memberships under user management.

- **Improvements**
  - Updated Redis configuration to support dynamic usernames from application properties.
  - Enhanced store status management using a command pattern for better extensibility.

- **Refactor**
  - Unified store open/close logic into a single service and controller method.
  - Removed the old membership reservation endpoint from its previous controller and relocated it for improved organization.

- **Bug Fixes**
  - Improved error responses for invalid or unsupported status values in API requests.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ybon1107 committed:
refactor: store - 1차 refactoring#73

- 불필요한 파일 삭제
- 상수 정의 및 활용

ybon1107 committed:
refactor: store - 1차 refactoring#73

- lua script 수정
- 서비스 위치 변경

ybon1107 committed:
refactor: store - 1차 refactoring#73

- store open/close command 패턴으로 활용
- 확장성 고려해 하나의 api로 통일

ybon1107 committed:
Merge pull request #21 from ybon1107/refactor/store/#73-store-refactoring

Refactor/store/#73 store refactoring

ybon1107 committed:
Merge pull request #231 from ybon1107/dev

store 1차 리팩토링

Leejieon commented on membership-service/src/main/java/com/boeingmerryho/business/membershipservice/exception/GlobalExceptionHandler.java:
Exception 세분화한거 너무 좋은 것 같습니다!

Leejieon reviewed (approved):
다시 또 리팩토링 진행해봅시다! 고생하셨어요!

ybon1107 committed:
refactor: store&membership - TODO 업데이트#73

- membership lua script 수정 및 부하 테스트 후 성능 비교
- store scheduler ZSET에서 String 으로 변경해 각 가게 및 날짜별 인덱싱 키 생성 방식으로 변경

ybon1107 committed:
Merge pull request #22 from ybon1107/refactor/store/#73-store-refactoring

refactor: store&membership - TODO 업데이트#73

ybon1107 committed:
Merge pull request #235 from ybon1107/dev

store&membership 피드백 토대로 TODO 추가

oneul0 reviewed (approved):
리펙토링 하시느라 고생많으셨습니다 👍
